doc/EnvironmentVariables.md
# Environment variables

List and overview of environment variables.

## ADAGUC_TRACE_TIMINGS

- TRUE: Measure timings like how much amount is spent on database queries, fileaccess and image generation
- FALSE: off.

The information is logged and made available in a header named `x-trace-timings`. 

## ADAGUC_ENABLELOGBUFFER

Controls behaviour of logstream buffer:
- FALSE (unset / default): unbuffered output with live logging, but can have a slower service as effect
- TRUE: buffered logging
- DISABLELOGGING: no logging at all

## ADAGUC_PATH

The root directory of the adaguc-server code base. Used to read data files like legend definitions and fonts. Defaults to `pwd`

## ADAGUC_DATASET_DIR

The directory for dataset configuration xml files. Defaults to `/data/adaguc-datasets`

## ADAGUC_DATA_DIR

The directory where data files are places, for example realtime updates of NetCDF files. Defaults to `/data/adaguc-data`

## ADAGUC_AUTOWMS_DIR

A sandbox directory where you can place your NetCDF, PNG, CSV or HDF5 files. A WMS service will be autogenerated and you can browse and preview the data and its containing variables. defaults to `/data/adaguc-autowms`

## ADAGUC_CONFIG

The location of the main server configuration file. Defaults to `${ADAGUC_PATH}/python/lib/adaguc/adaguc-server-config-python-postgres.xml`

## ADAGUC_NUMPARALLELPROCESSES

The amount of parallel processes the adaguc-server should use. Defaults to `4`

## ADAGUC_DB

The PostGres database configuration. Defaults to `"user=adaguc password=adaguc host=localhost `dbname=adaguc"

## ADAGUC_TMP

The temporary directory where adaguc-server can write temporary files.

###  ADAGUC_LOGFILE 

File where log messages will be stored


## ADAGUC_FONT 

A default TTF font location for print fonts in imagery. E.gFreeSans.ttf.

## ADAGUC_ONLINERESOURCE 

optional, specify the online resource in the CGI script itself, see [OnlineResource](configuration/OnlineResource.md) to configure in    

## PGBOUNCER_ENABLE 	

Enable or disable the usage of PostgreSQL connection pooling by PGBouncer. Default: true.

## PGBOUNCER_DISABLE_SSL 	

If PGBouncer is used, disable the usage of SSL. Default: true.

doc/Environment_Variables.md
Environment variables
=====================

The following environment variables allow you to change behaviour of the Adaguc server.

Some of these environment variables might be set via the `docker-compose.yml` file, see [Running adaguc-server with docker](/doc/Running.md)

| Environment variable | Description |
| -------------------- | ----------- |
| `ADAGUC_CONFIG` | pointer to the configuration file
| `ADAGUC_LOGFILE` | pointer where log messages should be stored, includes information logs and error logs
| `ADAGUC_ERRORFILE` | optional pointer which logs only error messages
| `ADAGUC_FONT` | Place where a TrueType font is stored, e.g. FreeSans.ttf
| `ADAGUC_DATARESTRICTION` | Optional pointer which controls access restrictions, by default set to FALSE, can be combinations of `ALLOW_WCS\|ALLOW_GFI\|ALLOW_METADATA\|SHOW_QUERYINFO`, separated with the \| token.<br /><br />`FALSE`: No restrictions (default, same as `ALLOW_WCS\|ALLOW_GFI\|ALLOW_METADATA`)<br>`ALLOW_WCS`: Allows the Web Coverage Service, download of data<br />`ALLOW_GFI`: Allows GetFeatureInfo requests, e.g. getting information about a certain location<br>`ALLOW_METADATA`: Allows getting NetCDF header metadata information<br>`SHOW_QUERYINFO`: When a query has failed, the corresponding query will be presented to the user. This feature is disabled by default.
| `ADAGUC_PATH` | optional, is used as variable susbstitution {ADAGUC_PATH} in the configuration files, should point to the adagucserver installation
| `ADAGUC_TMP` | optional, is used as variable susbstitution {ADAGUC_TMP} in the configuration files, location where tempfiles need to be written
| `ADAGUC_ONLINERESOURCE` | optional, specify the online resource in the CGI script itself, see [OnlineResource](/doc/configuration/OnlineResource.md) to configure in the xml file.
| `PGBOUNCER_ENABLE` | Enable or disable the usage of PostgreSQL connection pooling by PGBouncer. Default: `true`.
| `PGBOUNCER_DISABLE_SSL` | If PGBouncer is used, disable the usage of SSL. Default: `true`.
| `ADAGUC_DB` | The connection string used by PostgreSQL. Default: `host=adaguc-db port=5432 user=adaguc password=adaguc dbname=adaguc`.
| `ADAGUC_ENABLELOGBUFFER` | Enable or disable log buffering. If disabled, no debug logging is shown at all. Default: `true`.
| `ADAGUC_AUTOWMS_DIR` | Default: `/data/adaguc-autowms`
| `ADAGUC_DATA_DIR` | Default: `/data/adaguc-data`
| `ADAGUC_DATASET_DIR` | Default: `/data/adaguc-datasets`
| `ADAGUC_FORK_SOCKET_PATH` | Enables or disables the usage of an Adaguc fork server. If set to an adaguc-writable path, e.g. `adaguc.socket`, Adaguc will launch a server in the background and fork itself when handling requests. Communication will take place through this socket. If left empty or removed, Adaguc will launch a subprocess which comes with overhead.
| `ADAGUC_MAX_PROC_TIMEOUT` | Every request made to Adaguc will timeout after this many seconds. Default: `300` seconds.
| `ADAGUC_PORT` | Port to listen to for the webserver, used by `docker-compose.yml`. Default: port `443`.
| `EXTERNALADDRESS` | Adaguc-viewer and adaguc-explorer will be reachable on this hostname.
| `ADAGUCENV_RETENTIONPERIOD` | ISO 8601 time period string indicating how long files should be retained.
| `ADAGUCENV_ENABLECLEANUP` | Enable or disable cleaning.

### TODO:
- How/where you can apply these environment variables (.env, docker-compose.yaml, adaguc xml config)?
- What are the defaults?
- Which vars are optional or required?
- What is the format (string, caps only, path, etc)?

NEWS
Version 2.6.3 2021-11-30
    - HDF reader now reads variable length string attributes. These are usually written with h5py, using type H5T_C_S1.

Version 2.6.2 2021-11-22
    - Adjusted logging format, differentiating between acces log and application log.
Version 2.6.1 2021-11-09
    - CSV reader now detects floats, ints and strings properly with RegExp (thanks EduardoCastanho)

Version 2.6.0 2021-11-01
    - Moving to a python wrapper to run adaguc-server. For development with flask, for production with gunicorn.

Version 2.5.15 2021-09-22
    - The /data folder can now completely be mounted from another filesystem, all adaguc internal folders have moved to /adaguc.

Version 2.5.14 2021-09-17
    - WCS GetCoverage with Native parameters now returns data
    - PNG reader can extraxt bbox, time and crs from png text metadata properties

Version 2.5.13 2021-07-22
    - Extended logging options:
        - ADAGUC_ENABLELOGBUFFER=FALSE: unbuffered output, gives realtime logging. Unbuffered output can cause a slower service
        - ADAGUC_ENABLELOGBUFFER=TRUE: buffered logging
        - ADAGUC_ENABLELOGBUFFER=DISABLELOGGING: no logging at all

Version 2.5.11 2021-05-12
    - Correct rendering of IPCC data on North and South pole projections

Version 2.5.9 2021-05-11
    - Added tests for custom projection of Robinson, EPSG:3411 and EPSG:3413 for handling correct visualization of CMIP6 data for IPCC atlas by Predictia and Unican

Version 2.5.8 2021-04-07
    - Debug logging can be switched off from a dataset configuration

Version 2.5.7 2021-04-07
    - Contourlines can now be rendered without text

Version 2.5.4 2020-11-27
    - Improved drawing of contourlines for new climate normal maps.

Version 2.5.3 2020-11-18
    - More resilient reading of KNMI HDF5 files

Version 2.5.2 2020-11-13
    - Support to run Adaguc-server from python3.

Version 2.5.1 2020-11-13
    - Support for 32 contourline definitions (previousely 8)

Version 2.5.0 2020-11-13
    - Support CMAKE

Version 2.4.2 2020-11-09
    - Support for Python 3

Version 2.2.5 2019-10-18
    - Update db with specified path will now also use the filter specified in the layer.

Version 2.2.4 2019-09-25
    - AdditionalLayer functionality is Fixed (https://dev.knmi.nl/projects/adagucserver/wiki/AdditionalLayer)

Version 2.2.3 2019-09-24
    - The CDF Data model supports more NCML features to support DLR-PA VADUGS files out of the box. 
        In addition it is possible to add a time dimension using NCML. 
        Its value can be created from a global attribute using the timeValueFromGlobalAttribute.
    - Anydump has the commandline option -ncml to dump with the specified NCML file

Version 2.2.2 2019-09-10
    - Updatedb supports the --path parameter to scan files specified by an absolute path

Version 2.2.1 2019-08-27
    - adaguc-server has a new feature called getmetadata. It can dump a NetCDF file via the source parameter and return a json document containing the NetCDF metadata
    - Example: <adaguc host>/wms?source=<URLEncoded opendapURL>&service=metdata&request=getmetadata
    - adaguc-server supports reading PNG files accompanied by a info file, functional tests are added to demonstrate.

Version 2.2.0 2019-06-27

Docker/adaguc-viewer/config.js
var baseLayerConfiguration = [
  {
    service:"http://geoservices.knmi.nl/cgi-bin/bgmaps.cgi?",
    name:"naturalearth2",
    title:"World base layer Natural Earth ",
    enabled: true,
    type: 'twms',
    format:"image/gif"
  },{
    title:"arcGisCanvas",
    name:"arcGisCanvas",
    type: 'twms',
    enabled:false
  }, {
    title:"OSM",
    name:"OSM",
    type: 'twms',
    enabled:false
  },{
    service:"http://geoservices.knmi.nl/cgi-bin/bgmaps.cgi?",
    name:"streetmap",
    title:"Open streetmap",
    enabled: false,
    type: 'wms',
    format:"image/png"
  },{
    service:"http://geoservices.knmi.nl/cgi-bin/MODIS_Netherlands.cgi?",
    name:"modis_250m_netherlands_8bit",
    title:"Static Modis image of the Netherlands",
    enabled: false,
    format:"image/png"
  },{
    service:"http://geoservices.knmi.nl/cgi-bin/worldmaps.cgi?",
    name:"ne_10m_admin_0_countries_simplified",
    format:"image/png",
    title:"World country borders",
    enabled: true,
    keepOnTop:true
  },{
    service:"http://geoservices.knmi.nl/cgi-bin/worldmaps.cgi?",
    name:"grid1x1",
    format:"image/png",
    title:"Grid 1x1 degrees",
    enabled: false,
    keepOnTop:true
  },{
    service:"http://geoservices.knmi.nl/cgi-bin/worldmaps.cgi?",
    name:"grid2x2",
    format:"image/png",
    title:"Grid 2x2 degrees",
    enabled: false,
    keepOnTop:true
  },{
    service:"http://geoservices.knmi.nl/cgi-bin/worldmaps.cgi?",
    name:"grid5x5",
    format:"image/png",
    title:"Grid 5x5 degrees",
    enabled: false,
    keepOnTop:true
  },{
    service:"http://geoservices.knmi.nl/cgi-bin/worldmaps.cgi?",
    name:"grid10x10",
    format:"image/png",
    title:"Grid 10x10 degrees",
    enabled: false,
    keepOnTop:true
  },{
    service:"http://geoservices.knmi.nl/cgi-bin/worldmaps.cgi?",
    name:"nl_raster_latlon",
    format:"image/png",
    title:"KNMI baselayer of the Netherlands",
    enabled: false,
    keepOnTop:false
  },{
    service:"http://geoservices.knmi.nl/cgi-bin/worldmaps.cgi?",
    name:"world_country_names",
    format:"image/png",
    title:"Country names",
    enabled: false,
    keepOnTop:true
  }
];

var defaultProjection = {srs:'EPSG:4326',bbox:'-180,-90,180,90'};

var hashLocationNotfiyAddLayer = false;

var getFeatureInfoApplications = [
  {name:'Time series mode',iconCls:'button_getfeatureinfo',location:'apps/gfiapp_d3c3.html'}
  //,{name:'Glameps application',iconCls:'button_getfeatureinfo',location:'../gfiapps/GLAMEPS_gfiapp.html'}
];

var dataChooserConfiguration = [
  {
    title:'KNMI realtime precipitation radar',
    thumbnail:'img/knmi_radar_icon.png',
    service:'http://geoservices.knmi.nl/cgi-bin/RADNL_OPER_R___25PCPRR_L3.cgi?',
    layer:'RADNL_OPER_R___25PCPRR_L3_COLOR',
    srs:'EPSG:3857',
    bbox:'220000,6500000,1000000,7200000',
    baselayerservice:'http://geoservices.knmi.nl/cgi-bin/bgmaps.cgi?',
    baselayername:'streetmap',
    opacity:0.8
  },{
    title:'MSGCPP last 7 days',
    thumbnail:'img/750px-MSG-CPP_Screenshot-Google_Earth-cwp.png',
    service:'http://msgcpp-ogc-realtime.knmi.nl/msgrt.cgi?',
    layer:'atmosphere_optical_thickness_due_to_cloud'
  },{
    title:'MSGCPP archive',
    thumbnail:'img/750px-MSG-CPP_Screenshot-Google_Earth-cwp.png',
    service:'http://msgcpp-ogc-archive.knmi.nl/msgar.cgi?',
    layer:'atmosphere_optical_thickness_due_to_cloud'
  },/*{
    title:'OMI daily SO2',
    thumbnail:'http://adaguc.knmi.nl/contents/webservices/OMI_TDCSO2_L3.jpg',
    service:'http://geoservices.knmi.nl/cgi-bin/OMI___OPER_R___TDCSO2__L3.cgi?',
    layer:'OMI_L3_TDCSO2',
    srs:'EPSG:4326',
    bbox:'-180,-90,180,90'
  },*/{
    title:'OMI yearly NO2',
    thumbnail:'img/OMI_NO2_TYTRCNO_screenshot.png',
    service:'http://geoservices.knmi.nl/cgi-bin/OMI___OPER_R___TYTRCNO_L3.cgi?',
    layer:'omi_yearly_tropospheric_no2',
    srs:'EPSG:4326',
    bbox:'-180,-90,180,90'
  },/*{
    title:'Geoserver @ suite.opengeo.org',
    thumbnail:'http://suite.opengeo.org/geoserver/wms?service=WMS&version=1.3.0&request=GetMap&layers=usa:states&srs=EPSG:4326&bbox=24.956,-124.731,49.372,-66.97&format=image/png&width=780&height=330',
    service:'http://suite.opengeo.org/geoserver/wms?',
    layer:'usa:states'
  },*/{
    title:'Inspire->KNMI: Actuele 10min data',
    thumbnail:'http://geoservices.knmi.nl/cgi-bin/inspireviewservice.cgi?DATASET=urn:xkdc:ds:nl.knmi::Actuele10mindataKNMIstations/1/&SERVICE=WMS&&SERVICE=WMS&VERSION=1.3.0&REQUEST=GetMap&LAYERS=baselayer,ta,overlay&WIDTH=200&HEIGHT=150&CRS=EPSG%3A3857&BBOX=29109.947643979103,6500000,1190890.052356021,7200000&STYLES=temperature%2Fpoint&FORMAT=image/png&TRANSPARENT=TRUE&&time=current&showdims=true',
    service:'http://geoservices.knmi.nl/cgi-bin/inspireviewservice.cgi?DATASET=urn:xkdc:ds:nl.knmi::Actuele10mindataKNMIstations/1/&SERVICE=WMS&',
    layer:'ta'    
  },
  {
    title:'inspire:aardbevingen_nederland_2',
    thumbnail:'http://geoservices.knmi.nl/cgi-bin/inspire/aardbevingen_nederland_2.cgi?SERVICE=WMS&&SERVICE=WMS&VERSION=1.3.0&REQUEST=GetMap&LAYERS=baselayer,magnitude,overlay&WIDTH=200&HEIGHT=150&CRS=EPSG%3A3857&BBOX=29109.947643979103,6500000,1190890.052356021,7200000&STYLES=temperature%2Fpoint&FORMAT=image/png&TRANSPARENT=TRUE&&time=current&showdims=true',
    service:'http://geoservices.knmi.nl/cgi-bin/inspire/aardbevingen_nederland_2.cgi?SERVICE=WMS&',
    layer:'magnitude'    
  }
  ,
  {
    title:'inspire:aardbevingen_catalogus_1',
    thumbnail:'http://geoservices.knmi.nl/cgi-bin/inspire/aardbevingen_catalogus_1.cgi?SERVICE=WMS&&SERVICE=WMS&VERSION=1.3.0&REQUEST=GetMap&LAYERS=baselayer,magnitude&WIDTH=300&HEIGHT=200&CRS=EPSG%3A3857&BBOX=585870.0766630658,6555091.875607174,880402.9486353853,6744633.534542043&STYLES=magnitude_continous%2Fpoint&FORMAT=image/png&TRANSPARENT=TRUE&',
    service:'http://geoservices.knmi.nl/cgi-bin/inspire/aardbevingen_catalogus_1.cgi?',
    layer:'magnitude'    
  }
  
  ,{
    title:'Inspire->KNMI: Waarneem stations',
    thumbnail:'http://geoservices.knmi.nl/cgi-bin/inspireviewservice.cgi?DATASET=urn:xkdc:ds:nl.knmi::waarneemstations/2/&SERVICE=WMS&&SERVICE=WMS&VERSION=1.3.0&REQUEST=GetMap&LAYERS=baselayer,obs_temp,overlay&WIDTH=200&HEIGHT=150&CRS=EPSG%3A3857&BBOX=29109.947643979103,6500000,1190890.052356021,7200000&STYLES=temperature%2Fpoint&FORMAT=image/png&TRANSPARENT=TRUE&&time=current&showdims=true',
    service:'http://geoservices.knmi.nl/cgi-bin/inspireviewservice.cgi?DATASET=urn:xkdc:ds:nl.knmi::waarneemstations/2/&SERVICE=WMS&',
    layer:'obs_temp'    
  },
  {
    title:'Inspire->KNMI: Etmaalgegevens',
    thumbnail:'http://geoservices.knmi.nl/cgi-bin/inspireviewservice.cgi?DATASET=urn:xkdc:ds:nl.knmi::etmaalgegevensKNMIstations/1/&SERVICE=WMS&&SERVICE=WMS&VERSION=1.3.0&REQUEST=GetMap&LAYERS=baselayer,TG,overlay&WIDTH=200&HEIGHT=150&CRS=EPSG%3A3857&BBOX=29109.947643979103,6500000,1190890.052356021,7200000&STYLES=temperature%2Fpoint&FORMAT=image/png&TRANSPARENT=TRUE&time=current&showdims=true',
    service:'http://geoservices.knmi.nl/cgi-bin/inspireviewservice.cgi?DATASET=urn:xkdc:ds:nl.knmi::etmaalgegevensKNMIstations/1/&SERVICE=WMS&',
    layer:'TG'    
  },{
    title:'Harmonie forecasts',
    thumbnail:'http://geoservices.knmi.nl/cgi-bin/HARM_N25.cgi?SERVICE=WMS&&SERVICE=WMS&VERSION=1.3.0&REQUEST=GetMap&LAYERS=baselayer,precipitation_flux,overlay&WIDTH=200&HEIGHT=150&CRS=EPSG%3A3857&BBOX=143873.45679012343,6500000,1076126.5432098766,7200000&STYLES=precip_cwk%2Fbilinearcontour&FORMAT=image/png&TRANSPARENT=TRUE&showdims=true',
    service:'http://geoservices.knmi.nl/cgi-bin/HARM_N25.cgi?',
    layer:'precipitation_flux'    

    
  },{
    title:'MSG Seviri Ash products',
    thumbnail:'http://geoservices.knmi.nl/cgi-bin/SEVIR_OPER_R___VOLE____L2.cgi?SERVICE=WMS&&SERVICE=WMS&VERSION=1.3.0&REQUEST=GetMap&LAYERS=baselayer,MSG3/ash_loading,overlay&WIDTH=400&HEIGHT=300&CRS=EPSG%3A4326&BBOX=-74.36030201,-123.8533563909424,74.88885300999999,123.8533563909424&STYLES=ash_loading%2Fnearest&FORMAT=image/png&TRANSPARENT=TRUE&&timecurrent&title=MSG%20Seviri%20Ash',
    service:'http://geoservices.knmi.nl/cgi-bin/SEVIR_OPER_R___VOLE____L2.cgi',
    layer:'MSG3/ash_loading'    

    
  },{
        title:'Accidents in traffic',
        thumbnail:'img/accidents_weather.jpg',
        service:'http://geoservices.knmi.nl/cgi-bin/weathertraffic.cgi',
        layer:'Loctypon'
  },
  {
        title:'Himawari Demo',
        thumbnail:'http://geoservices.knmi.nl/cgi-bin/himawari.cgi?SERVICE=WMS&&SERVICE=WMS&VERSION=1.3.0&REQUEST=GetMap&LAYERS=tiled&WIDTH=400&HEIGHT=300&CRS=EPSG%3A4326&BBOX=-31.97397264884439,80.83215795209837,13.546051112849678,156.38109791260632&STYLES=auto%2Frgba&FORMAT=image/png&TRANSPARENT=TRUE&&time=2016-11-14T07%3A50%3A00Z',
        service:'http://geoservices.knmi.nl/cgi-bin/himawari.cgi',
        layer:'tiled'

  }
  
  
  
  
];

var mapTypeConfiguration = [ 
  { 
    title: 'World WGS84', 
    bbox: [-180,-90,180,90], 
    srs: 'EPSG:4326',
    baselayer:{service:'http://geoservices.knmi.nl/cgi-bin/bgmaps.cgi?',name:'naturalearth2',type: 'wms'}
  },/*{ 
    title: 'Mollweide (7399)', 
    bbox: [-18157572.744146045,-11212941.682924412,18085661.018022258,11419683.192411266],
    srs: 'EPSG:7399',
    baselayer:{service:'http://geoservices.knmi.nl/cgi-bin/bgmaps.cgi?',name:'naturalearth2',type: 'wms'}
  },*/
  { 
    title: 'Robinson', 
    bbox: [-17036744.451383516,-10711364.114367772,16912038.081015453,10488456.659686875],
    srs: 'EPSG:54030',
    baselayer:{service:'http://geoservices.knmi.nl/cgi-bin/bgmaps.cgi?',name:'naturalearth2',type: 'wms'}
  },{ 
    title: 'World Mercator', 
    bbox: [-19000000,-19000000,19000000,19000000],
    srs: 'EPSG:3857',
    baselayer:{service:'http://geoservices.knmi.nl/cgi-bin/bgmaps.cgi?',name:'naturalearth2',type: 'wms'}
  },{ 
    title: 'Openstreetmap', 
    bbox: [-19000000,-19000000,19000000,19000000],   
    srs: 'EPSG:3857', 
    baselayer:{service:'http://geoservices.knmi.nl/cgi-bin/bgmaps.cgi?',name:'streetmap',type: 'wms'}
  },{ 
    title: 'Northern Hemisphere', 
    bbox: [-5661541.927991125,-3634073.745615984,5795287.923063262,2679445.334384017],
    srs: 'EPSG:3411',
    baselayer:{service:'http://geoservices.knmi.nl/cgi-bin/bgmaps.cgi?',name:'naturalearth2',type: 'wms'}
  },{ 
    title: 'Southern Hemisphere', 
    bbox: [-4589984.273212382,-2752857.546211313,5425154.657417289,2986705.2537886878],
    srs: 'EPSG:3412',
    baselayer:{service:'http://geoservices.knmi.nl/cgi-bin/bgmaps.cgi?',name:'naturalearth2',type: 'wms'}
  },{ 
    title: 'Europe North Pole', 
    bbox: [-13000000,-13000000,13000000,13000000],
    srs: 'EPSG:3575',
    baselayer:{service:'http://geoservices.knmi.nl/cgi-bin/bgmaps.cgi?',name:'naturalearth2',type: 'wms'}
  },
  
  
  
 
  
  { 
    title: 'Europe stereographic', 
    bbox: [-2776118.977564746,-6499490.259201691,9187990.785775745,971675.53185069], 
    srs: 'EPSG:32661',
    baselayer:{service:'http://geoservices.knmi.nl/cgi-bin/bgmaps.cgi?',name:'naturalearth2',type: 'wms'}
  }, { 
    title: 'North America', 
    bbox: [-2015360.8716608454,-697107.5349683464,9961718.159421016,6782157.107682772],
    srs: 'EPSG:50001',
    baselayer:{service:'http://geoservices.knmi.nl/cgi-bin/bgmaps.cgi?',name:'naturalearth2',type: 'wms'}
  },
  
  { 
    title: 'Openstreetmap NL', 
    bbox: [220000,6500000,1000000,7200000],   
    srs: 'EPSG:3857', 
    baselayer:{service:'http://geoservices.knmi.nl/cgi-bin/bgmaps.cgi?',name:'streetmap',type: 'wms'}
  }/*,{ 
    title: 'The Netherlands (28992)', 
    bbox: [-350000,125000,700000,900000 ],
    srs: 'EPSG:28992',
    baselayer:{service:'http://geoservices.knmi.nl/cgi-bin/bgmaps.cgi?',name:'naturalearth2',type: 'wms'}
  },*//*{ 
    title: 'PDOK BRT NL', 
    bbox: [-350000,125000,700000,900000],   
    srs: 'EPSG:28992', 
    baselayer:{service:'http://geoservices.knmi.nl/cgi-bin/bgmaps.cgi?',name:'streetmap',type: 'wms'}
  },*//*{ 
    title: 'Openstreetmap Schiphol', 
    bbox: [515813.2988839851,6850485.5537392385,544160.8241851525,6869318.449956324],   
    srs: 'EPSG:3857', 
    baselayer:{service:'http://geoservices.knmi.nl/cgi-bin/bgmaps.cgi?',name:'streetmap',type: 'wms'}
  }*/
  
  /*,{ 
    title: 'Schiphol Satellite + Aerial Mercator', 
    bbox: [522402.16196646384,6852917.910540686,537831.8792436487,6863959.075064662],
    srs: 'EPSG:3857',
    baselayer:{service:'http://birdexp03.knmi.nl/cgi-bin/plieger/wmst.cgi?',name:'satellite'}
  },{ 
    title: 'Topographic Mercator in NL', 
    bbox: [220000,6500000,1000000,7200000],
    srs: 'EPSG:3857',
    baselayer:{service:'http://birdexp03.knmi.nl/cgi-bin/plieger/wmst.cgi?',name:'topo'}
  },{ 
    title: 'Canvas Mercator in NL', 
    bbox: [220000,6500000,1000000,7200000],
    srs: 'EPSG:3857',
    baselayer:{service:'http://birdexp03.knmi.nl/cgi-bin/plieger/wmst.cgi?',name:'canvas'}
  }*/
];

var defaultLanguage = { language: 'en' }; // <-- Language for the ADAGUC viewer.
var defaultUsernameSearch = "adaguc"; // <-- Username for the GeoNames API. 1
var geoNamesURL = "http://api.geonames.org/search?q={searchTerm}&username={username}&maxRows=1"; // <-- URL for the GeoNames API. Requires 'defaultUsernameSearch'    
      
//FOR JSP:
/*
var scaleBarURL        = "http://webgis.nmdc.eu/viewer2.0/webmapjs/php/makeScaleBar.php?";
var xml2jsonrequestURL = "/impactportal/AdagucViewer?SERVICE=XML2JSON&";
var requestProxy = "/impactportal/AdagucViewer?SERVICE=PROXY&";
*/

//For PHP:

var requestProxy       = "webmapjs_php/MakeRequest.php?";
var xml2jsonrequestURL = "webmapjs_php/xml2jsonrequest.php?"

// xml2jsonrequestURL = 'http://localhost:8080/adaguc-services/xml2json?'
getFeatureInfoApplications.push({name:'AutoWMS',iconCls:'button_getfeatureinfo'});
getFeatureInfoApplications.open = 'AutoWMS';


var WMJSTileRendererTileSettings = {
  arcGisCanvas: {
    'EPSG:3857': {
      home: 'http://services.arcgisonline.com/ArcGIS/rest/services/Canvas/World_Light_Gray_Base/MapServer/tile/',
      minLevel: 1,
      maxLevel: 16,
      tileServerType: 'arcgisonline',
      copyRight: 'Basemap copyright: 2013 Esri, DeLorme, NAVTEQ'
    },
    'EPSG:28992': {
      home: 'http://services.arcgisonline.nl/arcgis/rest/services/Basiskaarten/Canvas/MapServer/tile/',
      minLevel: 1,
      maxLevel: 12,
      origX:-285401.92,
      origY:903401.92,
      resolution:3440.64,
      tileServerType: 'arcgisonline',
      copyRight: 'Basiskaart bronnen: Esri Nederland, Esri, Kadaster, CBS en Rijkswaterstaat'
    }
  },
  arcGisTopo: {
    'EPSG:3857': {
      home: 'http://services.arcgisonline.com/ArcGIS/rest/services/World_Topo_Map/MapServer/tile/',
      minLevel: 1,
      maxLevel: 19,
      tileServerType: 'arcgisonline',
      copyRight: 'Basemap sources: Esri, DeLorme, NAVTEQ, TomTom, Intermap, increment P Corp., GEBCO, USGS, FAO, NPS, NRCAN, GeoBase, ' +
        'IGN, Kadaster NL, Ordnance Survey, Esri Japan, METI, Esri China (Hong Kong), and the GIS User Community'
    },
    'EPSG:28992': {
      home: 'http://services.arcgisonline.nl/arcgis/rest/services/Basiskaarten/Topo/MapServer/tile/',
      minLevel: 1,
      maxLevel: 12,
      origX:-285401.92,
      origY:903401.92,
      resolution:3440.64,
      tileServerType: 'arcgisonline',
      copyRight: 'Basiskaart bronnen: Esri Nederland, Esri, Kadaster, CBS, Min VROM, Rijkswaterstaat en gemeenten: Rotterdam, Breda, Tilburg'
    }
  },
  arcGisOceanBaseMap:  {
    home: 'http://services.arcgisonline.com/ArcGIS/rest/services/Ocean_Basemap/MapServer/tile/',
    minLevel: 1,
    maxLevel: 19,
    tileServerType: 'arcgisonline',
    copyRight: 'Basemap sources: Esri, GEBCO, NOAA, National Geographic, DeLorme, NAVTEQ, Geonames.org, and other contributors'
  },
  arcGisSat: {
    'EPSG:4326': {
      home: 'http://services.arcgisonline.com/ArcGIS/rest/services/ESRI_Imagery_World_2D/MapServer/tile/',
      minLevel: 1,
      maxLevel: 15,
      tileServerType: 'arcgisonline',
      origX:-180,
      origY:90,
      resolution:0.3515625,
      tileSize:512
    },
    'EPSG:3857': {
      home: 'http://services.arcgisonline.com/ArcGIS/rest/services/World_Imagery/MapServer/tile/',
      minLevel: 1,
      maxLevel: 18,
      tileServerType: 'arcgisonline'
    }
  },
  OSM: {
    'EPSG:3857': {
      home: '//b.tile.openstreetmap.org/',
      minLevel: 1,
      maxLevel: 16,
      tileServerType: 'osm'
    },
    'EPSG:28992': {
      home: 'http://services.arcgisonline.nl/ArcGIS/rest/services/Basiskaarten/PDOK_BRT/MapServer/tile/',
      minLevel: 1,
      maxLevel: 12,
      origX:-285401.92,
      origY:903401.92,
      resolution:3440.64,
      tileServerType: 'arcgisonline',
      copyRight: 'Basiskaart bronnen: PDOK, Kadaster, OpenStreetMap'
    }
  },
  NaturalEarth2: {
    'EPSG:3411': {
      home: 'http://geoservices.knmi.nl/tiledbasemaps/NaturalEarth2/EPSG3411/',
      minLevel: 1,
      maxLevel: 6,
      origX:-12400000,
      origY:12400000,
      resolution:96875,
      tileServerType: 'wmst',
      copyRight: 'NPS - Natural Earth II'
    },
    'EPSG:3412': {
      home: 'http://geoservices.knmi.nl/tiledbasemaps/NaturalEarth2/EPSG3412/',
      minLevel: 1,
      maxLevel: 6,
      origX:-12400000,
      origY:12400000,
      resolution:96875,
      tileServerType: 'wmst',
      copyRight: 'NPS - Natural Earth II'
    },
    'EPSG:3575': {
      home: 'http://geoservices.knmi.nl/tiledbasemaps/NaturalEarth2/EPSG3575/',
      minLevel: 1,
      maxLevel: 6,
      origX:-13000000,
      origY:13000000,
      resolution:101562.5,
      tileServerType: 'wmst',
      copyRight: 'NPS - Natural Earth II'
    },
    'EPSG:3857': {
      home: 'http://geoservices.knmi.nl/tiledbasemaps/NaturalEarth2/EPSG3857/',
      minLevel: 1,
      maxLevel: 7,
      tileServerType: 'wmst',
      copyRight: 'NPS - Natural Earth II'
    },
    'EPSG:4258': {
      home: 'http://geoservices.knmi.nl/tiledbasemaps/NaturalEarth2/EPSG4326/',
      minLevel: 1,
      maxLevel: 6,
      origX:-180,
      origY:90,
      resolution:0.703125,
      tileServerType: 'wmst',
      copyRight: 'NPS - Natural Earth II'
    },
    'EPSG:4326': {
      home: 'http://geoservices.knmi.nl/tiledbasemaps/NaturalEarth2/EPSG4326/',
      minLevel: 1,
      maxLevel: 6,
      origX:-180,
      origY:90,
      resolution:0.703125,
      tileServerType: 'wmst',
      copyRight: 'NPS - Natural Earth II'
    },
    'EPSG:28992': {
      home: 'http://geoservices.knmi.nl/tiledbasemaps/NaturalEarth2/EPSG28992/',
      minLevel: 1,
      maxLevel: 5,
      origX:-2999000,
      origY:2995500,
      resolution:23437,
      tileServerType: 'wmst',
      copyRight: 'NPS - Natural Earth II'
    },
    'EPSG:32661': {
      home: 'http://geoservices.knmi.nl/tiledbasemaps/NaturalEarth2/EPSG32661/',
      minLevel: 1,
      maxLevel: 7,
      origX:-5000000.0,
      origY:4000000.0,
      resolution:58593.75,
      tileServerType: 'wmst',
      copyRight: 'NPS - Natural Earth II'
    },
    'EPSG:54030': {
      home: 'http://geoservices.knmi.nl/tiledbasemaps/NaturalEarth2/EPSG54030/',
      minLevel: 1,
      maxLevel: 7,
      origX:-17000000.0,
      origY:8625830.0,
      resolution:132812.5,
      tileServerType: 'wmst',
      copyRight: 'NPS - Natural Earth II'
    }
  }
};

